core/src/scene/drawRule.cpp

```diff
@@ -203,6 +222,20 @@ void DrawRuleMergeSet::apply(const Feature& _feature, const SceneLayer& _layer,
         }
 
         if (valid) {
+
+            // build outline explicitly with outline style
+            const auto& outlineStyleName = rule.findParameter(StyleParamKey::outline_style);
+            if (outlineStyleName) {
+                auto& styleName = outlineStyleName.value.get<std::string>();
+                auto* outlineStyle = _builder.getStyleBuilder(styleName);
+                if (!outlineStyle) {
+                    LOGE("Invalid style %s", styleName.c_str());
+                } else {
+                    outlineStyle->addFeature(_feature, rule.outlineRule());
```

Comment:
I'm not sure we need to create a copy of the original rule here, could we just do something like:
```
rule.isOutlineOnly = true;
outlineStyle->addFeature(_feature, rule);
rule.isOutlineOnly = false;
```